### PR TITLE
MEN-3880 CI pipeline added

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Review by alfrunes — APPROVED** [review]

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    - apt-get update && apt-get install -yyq liblzma-dev libssl-dev
+
+  script:
+    # Test if code was formatted with 'go fmt'
+    # Command will format code and return modified files
+    # fail if any have been modified.
+    - if [ -n "$(go fmt)" ]; then echo 'Code is not formatted with "go fmt"'; false; fi
+
+    # Perform static code analysys
+    - go vet `go list ./... | grep -v vendor`
```
Comment (oleorhagen) [inline .gitlab-ci.yml:43]: I'm pretty sure the inverse grep is not needed. `go list ./...` should ignore vendor by default.

**Review by oleorhagen — COMMENTED** [review]
I'm curious why we're not using: https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-golang-static.yml here? 
Was it considered?
It'd be nice to keep CI setups in one place (makes version upgrades a lot easier).

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
+include:
+  - project: 'Northern.tech/Mender/mendertesting'
+    file: '.gitlab-ci-github-status-updates.yml'
+
+stages:
+  - test
+  - publish
+
+test:unit:
```
Comment (lluiscampos) [inline .gitlab-ci.yml:9]: I am very late here (I was not subscribed to this repo by mistake), but we could have done all this with the `.gitlab-ci-check-golang-unittests.yml` template. What do we have in this file that differs from the template?

**Comment by lluiscampos** [comment]
> I'm curious why we're not using: https://github.com/mendersoftware/mendertesting/blob/master/.gitlab-ci-check-golang-static.yml here?
> Was it considered?
> It'd be nice to keep CI setups in one place (makes version upgrades a lot easier).

+1

**Pull request merged**